knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -51,6 +55,36 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for Kourier resources.
+	manifest, err := kourier.RawManifest(ingressNamespace, mgr.GetClient())
+	if err != nil {
+		return err
+	}
+	resources := manifest.Resources()
+
+	gvkToKourier := make(map[schema.GroupVersionKind]runtime.Object)
+	for _, u := range resources {
+		resource := u
```

Comment:
This seems redundant? Is it needed?

Reply:
Yes, it is needed. Otherwise, `gvkToKourier[u.GroupVersionKind()]`  gets same address for the value.

Reply:
If you want to actually get the ref to the slice you'll need to do

```golang
for i := range resources {
      resource := &resources[i]
  		gvkToKourier[resource.GroupVersionKind()] = resource
}
```

Reply:
Okay, I changed it.